File: app/src/main/java/com/icostel/facerecognition/di/OnActivityCreateLifecycleCallback.java
package com.icostel.facerecognition.di;

import android.app.Activity;
import android.app.Application;
import android.os.Bundle;

public interface OnActivityCreateLifecycleCallback extends Application.ActivityLifecycleCallbacks {
    @Override
    default void onActivityCreated(Activity activity, Bundle savedInstanceState) {
        handleOnCreate(activity, savedInstanceState);
    }

    @Override
    default void onActivityStarted(Activity activity) { }

    @Override
    default void onActivityResumed(Activity activity) { }

    @Override
    default void onActivityPaused(Activity activity) { }

    @Override
    default void onActivityStopped(Activity activity) { }

    @Override
    default void onActivitySaveInstanceState(Activity activity, Bundle outState) { }

    @Override
    default void onActivityDestroyed(Activity activity) { }

    void handleOnCreate(Activity activity, Bundle savedInstanceState);
}

File: app/src/main/java/com/icostel/facerecognition/di/OnActivityCreateLifecycleCallback.kt
package com.icostel.facerecognition.di

import android.app.Activity
import android.app.Application
import android.os.Bundle

interface OnActivityCreateLifecycleCallback : Application.ActivityLifecycleCallbacks {
    override fun onActivityCreated(activity: Activity, savedInstanceState: Bundle) {
        handleOnCreate(activity, savedInstanceState)
    }

    override fun onActivityStarted(activity: Activity) {}

    override fun onActivityResumed(activity: Activity) {}

    override fun onActivityPaused(activity: Activity) {}

    override fun onActivityStopped(activity: Activity) {}

    override fun onActivitySaveInstanceState(activity: Activity, outState: Bundle) {}

    override fun onActivityDestroyed(activity: Activity) {}

    fun handleOnCreate(activity: Activity, savedInstanceState: Bundle)
}
